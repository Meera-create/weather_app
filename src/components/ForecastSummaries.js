import React from 'react';
import ForecastSummary from './ForecastSummary';
import '../styles/ForecastSummaries.css'


function ForecastSummaries({forecasts, onForecastSelect}){
    //console.log(forecasts,"EFJNFNJKESNFJKS")
    return (
        <div className="forecast-summaries">
        {forecasts.map(forecast => (
            <ForecastSummary
            key={forecast.date}
            date={forecast.date}
            description={forecast.description}
            icon={forecast.icon.toString()}
            onSelect={onForecastSelect}
            temperature={forecast.temperature}
            />

        ))}
        </div>
       
    );
};

export default ForecastSummaries;

//props here is an array of forecasts from forecast.json