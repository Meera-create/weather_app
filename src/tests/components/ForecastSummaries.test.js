import React from  'react'
import {render, getByTestId} from "@testing-library/react"
import ForecastSummaries from '../../components/ForecastSummaries'

describe("Forecast summaries",()=>{
    const validProps2=[
        {
            temperature : {
                min: 22,
                max:35,
            },
            icon :'StubIcon',
            description :'StubDescription' ,
            date :111111
        },
        {
            temperature : {
                min: 12,
                max:30,
            },
            icon :'StubIcon2',
            description :'StubDescription2' ,
            date :222222
        },
    
    ];
   

it("renders the summary components correctly",()=>{
    const{asFragment} = render(
        <ForecastSummaries forecasts={validProps2} />)         
expect(asFragment()).toMatchSnapshot();
});

it("renders forecast summary the correct number of times",()=>{
    const {getAllByTestId}=render(
        <ForecastSummaries forecasts={validProps2} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2)

    })

})