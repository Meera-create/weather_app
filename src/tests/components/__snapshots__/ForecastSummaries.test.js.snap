// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast summaries renders the summary components correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        111111
      </div>
      <div
        class="forecast-summary_description"
      >
        StubDescription
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        StubIcon
      </div>
      <div
        class="forecast-summary_temperature"
      >
        35°C
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        222222
      </div>
      <div
        class="forecast-summary_description"
      >
        StubDescription2
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        StubIcon2
      </div>
      <div
        class="forecast-summary_temperature"
      >
        30°C
      </div>
    </div>
  </div>
</DocumentFragment>
`;
